### PR TITLE
feat(770): Add schema for trigger table

### PR DESCRIPTION
**Pull request by d2lam**
Add schema for the new trigger table.
No endpoint should be exposed to the users.

Related: https://github.com/screwdriver-cd/screwdriver/issues/770

### CODE REVIEW

File: models/trigger.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        .example(12345),
+
+    pipelineId: Joi
+        .number().integer().positive()
+        .example(56789),
+
+    jobName: Job.jobname
+        .example('component'),
+
+    trigger: Joi
```
Comment (minz1027) [inline models/trigger.js]: Can we user the EXTERNAL_TRIGGER we just added to the regex? Just to be consistent

File: models/trigger.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
```
Comment (minz1027) [inline models/trigger.js]: `trigger` is a reserved keyword in a lot of SQL database. should probably user something else as the column name

Comment (d2lam) [reply models/trigger.js]: Changed the column name to `start` based on your suggestion :)

File: models/trigger.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+const Regex = require('../config/regex');
+
+const MODEL = {
+    id: Joi
+        .number().integer().positive()
+        .example(12345),
+
+    src: Joi
+        .string().regex(Regex.EXTERNAL_TRIGGER)
+        .example('~sd@1234:component'),
```
Comment (jithin1987) [inline models/trigger.js]: why is `~ad@` syntax required here ? can't it be just `pipeline:jobName` format ?

Comment (tkyi) [reply models/trigger.js]: I think we're keeping `~sd@` for simplicity so we can reuse the `EXTERNAL_TRIGGER` regex to parse the `src` and `dest`.

Comment (minz1027) [reply models/trigger.js]: And I feel it's more consistent this way. Just in case we change the format we want for the EXTERNAL_TRIGGER later.

Comment (jithin1987) [reply models/trigger.js]: Sounds good.

**Pull request merged**